File: eMaestroD.NgUI/src/app/Manage/Models/product-view-model.ts
export interface ProductViewModel {

  prodID?: number | null;
  prodGrpID?: number | null;
  prodCode?: string | null;
  shortName?: string | null;
  prodName?: any | null;
  descr?: string | null;
  prodUnit?: string | null;
  unitQty?: number | null;
  purchRate?: number | null;
  sellRate?: number | null;
  isTaxable?: boolean | null;
  isStore?: boolean | null;
  isRaw?: boolean | null;
  minQty?: number | null;
  maxQty?: number | null;
  mega?: string | null;
  active?: boolean | null;
  crtBy?: string | null;
  crtDate?: Date | null;
  modBy?: string | null;
  modDate?: Date | null;
  comID?: number | null;
  retailPrice?: number | null;
  tp?: number | null;
  isBonus?: boolean | null;
  isDiscount?: boolean | null;
  prodManuID?: number | null;
  depID?: number | null;
  categoryID?: number | null;
  isImported?: boolean | null;
  prodSize?: string | null;
  prodColor?: string | null;

  // ProductBarCodes Table Fields
  prodBCID?: number | null;
  barCode?: string | null;
  costPrice?: number | null;
  salePrice?: number | null;
  tradePrice?: number | null;
  unit?: string | null;
  fobPrice?: number | null;

  prodManuName?: string | null;
  depName?: string | null;
  categoryName?: string | null;
  prodGrpName?: string | null;

  lastCostPrice? : number | null;
  currentStock? : number | null;
  avgUnitPrice? : number | null;



  qty?: number;
  qtyBal?: number;
  discount?: number;
  bonusQty?: number;
  discountAmount?: number;
  discountPercent?: number;
  taxPercent?: number;
  taxAmount?: number;
  extraDiscountPercent?: number;
  extraDiscountAmount?: number;
  advanceTaxPercent?: number;
  advanceTaxAmount?: number;
  extraAdvanceTaxAmount?: number;
  extraAdvanceTaxPercent?: number;
  rebatePercent?: number;
  rebateAmount?: number;
  grossValue?: number;
  discountedGross?: number;
  netAmountBeforeRebate?: number;
  netAmount?: number;
  netRate?: number;
  amount?: string;
  batchNo?:string;
  notes?:string;
  expiryDate?:Date;
  prodInvoiceID?:number | 0;
}

File: Models/VMModels/ProductViewModel.cs
using eMaestroD.Models.Custom;
using eMaestroD.Models.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Models.VMModels
{
    public class ProductViewModel : IEntityBase
    {
        [DisplayName(Name = "Product Barcode")]
        public string? barCode { get; set; }
        [DisplayName(Name = "Product Name")]
        public string prodName { get; set; }
        [DisplayName(Name = "Manufacture")]
        public string? prodManuName { get; set; }
        [DisplayName(Name = "Department")]
        public string? depName { get; set; }
        [DisplayName(Name = "Category")]
        public string? categoryName { get; set; }
        [DisplayName(Name = "Brand")]
        public string? prodGrpName { get; set; }
        [DisplayName(Name = "Purchase Rate")]
        public decimal? purchRate { get; set; }
        [DisplayName(Name = "Sell Rate")]
        public decimal? sellPrice { get; set; }
        [DisplayName(Name = "Current Stock")]
        public decimal? currentStock { get; set; }
        [DisplayName(Name = "Min Qty")]
        public decimal? minQty { get; set; }
        [DisplayName(Name = "Max Qty")]
        public decimal? maxQty { get; set; }
        [DisplayName(Name = "Size")]
        public string? prodSize { get; set; }
        [DisplayName(Name = "Color")]
        public string? prodColor { get; set; }
        public bool? isTaxable { get; set; }
        public bool? isImported { get; set; }

        [HiddenOnRender]
        public int? prodID { get; set; }
        [HiddenOnRender]
        public int? prodGrpID { get; set; }
        [HiddenOnRender]
        public string prodCode { get; set; }
        [HiddenOnRender]
        public string? shortName { get; set; }
        [HiddenOnRender]
        public string descr { get; set; }
        [HiddenOnRender]
        public string prodUnit { get; set; }
        [HiddenOnRender]
        public decimal? unitQty { get; set; }

        [HiddenOnRender]
        public decimal? sellRate { get; set; }

        [HiddenOnRender]
        public bool? isStore { get; set; }
        [HiddenOnRender]
        public bool? isRaw { get; set; }
        [HiddenOnRender]
        public string? mega { get; set; }
        [HiddenOnRender]
        public bool? active { get; set; }
        [HiddenOnRender]
        public string? crtBy { get; set; }
        [HiddenOnRender]
        public DateTime? crtDate { get; set; }
        [HiddenOnRender]
        public string? modBy { get; set; }
        [HiddenOnRender]
        public DateTime? modDate { get; set; }
        [HiddenOnRender]
        public int? comID { get; set; }
        [HiddenOnRender]
        public decimal? retailPrice { get; set; }
        [HiddenOnRender]
        public decimal? tp { get; set; }
        [HiddenOnRender]
        public bool? isBonus { get; set; }
        [HiddenOnRender]
        public bool? isDiscount { get; set; }
        [HiddenOnRender]
        public int? prodManuID { get; set; }
        [HiddenOnRender]
        public int? depID { get; set; }
        [HiddenOnRender]
        public int? categoryID { get; set; }
      

        // ProductBarCodes Table Fields

        [Key]
        [HiddenOnRender]
        public int? prodBCID { get; set; }
  
        [HiddenOnRender]
        public decimal? costPrice { get; set; }
        [HiddenOnRender]
        public decimal? salePrice { get; set; }
        [HiddenOnRender]
        public decimal? tradePrice { get; set; }
        [HiddenOnRender]
        public string? unit { get; set; }
        [HiddenOnRender]
        public decimal? fobPrice { get; set; }
        [HiddenOnRender]
        public decimal? lastCost { get; set; }
        [HiddenOnRender]
        public int? vendID { get; set; }
        [HiddenOnRender]
        public string? vendName { get; set; }
        [HiddenOnRender]
        public int? preference { get; set; }
        [HiddenOnRender]
        public decimal? sharePercentage { get; set; }
    }
}
